### ISSUE TITLE
relax federated type validation rules

### ISSUE THREAD

**Issue by dariuszkuc** [ExpediaGroup/graphql-kotlin#1566]
### Discussed in https://github.com/ExpediaGroup/graphql-kotlin/discussions/1565

<div type='discussions-op-text'>

<sup>Originally posted by **hchen** September 29, 2022</sup>
Hi team, we are getting the following errors when trying to use  `@ExternalDirective` and `@RequiresDirective` in base type.

```
com.expediagroup.graphql.generator.federation.exception.InvalidFederatedSchema: Invalid federated schema:
 - base Product type has fields marked with @requires directive, validatedField=shippingEstimate
 - base Product type has fields marked with @external directive, fields=[weight]
```

We want to caculate `shippingEstimate` of the `Product` which relies on the `weight` field. This weight field is resolved by another subgraph.

I'm wondering why these checks are added and if this can be disabled using a config.

Thanks in advance for your help.

</div>

**Comment by hchen** [comment]
@dariuszkuc thanks for tracking the request. 
Just want to add that there is another validation to be removed in `validateFieldSet.kt` file.

```
            errors.add("$errorMessage specifies invalid field set - type incorrectly references external field=${targetField.name}")

```

**Comment by hchen** [comment]
@dariuszkuc I'm wondering what's the ETA for this fix to be released?

**Comment by hchen** [comment]
@dariuszkuc I have submitted a PR to relax the validation rules. Would appreciate if you can review when you get a chance!

**Comment by dariuszkuc** [comment]
resolved in #1571
